packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -259,29 +266,14 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
         // INVARIANT: The `msg.value` must exactly equal the required bond.
         if (getRequiredBond(nextPosition) != msg.value) revert IncorrectBondAmount();
 
-        // Fetch the grandparent clock, if it exists.
-        // The grandparent clock should always exist unless the parent is the root claim.
-        Clock grandparentClock;
-        if (parent.parentIndex != type(uint32).max) {
-            grandparentClock = claimData[parent.parentIndex].clock;
-        }
-
         // Compute the duration of the next clock. This is done by adding the duration of the
         // grandparent claim to the difference between the current block timestamp and the
         // parent's clock timestamp.
-        Duration nextDuration = Duration.wrap(
-            uint64(
-                // First, fetch the duration of the grandparent claim.
-                grandparentClock.duration().raw()
-                // Second, add the difference between the current block timestamp and the
-                // parent's clock timestamp.
-                + block.timestamp - parent.clock.timestamp().raw()
-            )
-        );
+        Duration nextDuration = getChallengerDuration(_challengeIndex);
 
         // INVARIANT: A move can never be made once its clock has exceeded `GAME_DURATION / 2`
         //            seconds of time.
-        if (nextDuration.raw() > GAME_DURATION.raw() >> 1) revert ClockTimeExceeded();
+        if (nextDuration.raw() == GAME_DURATION.raw() >> 1) revert ClockTimeExceeded();
```

Comment:
This is inconsistent with the INVARIANT comment above:
```
A move can never be made once its clock has exceeded `GAME_DURATION / 2`
        //            seconds of time.
```
Why are we changing the behavior?

Reply:
See where the `nextDuration` is fetched. The code maxes out the clock at `GAME_DURATION.raw() >> 1`, and does not allow it to continue running. This is easier to reason about than allowing the clocks to continue ticking imo. There's a set max time on them, and past that, we don't continue counting :)

Reply:
It's not inconsistent because the `getChallengerDuration` method returns a **maximum** of `GAMEDURATION.raw() >> 1`.

See: https://github.com/ethereum-optimism/optimism/blob/inphi/fix-rez/packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol#L661-L665

Reply:
What I mean is that until now, we don't consider clocks that have elapsed for exactly `GAME_DURATION >> 1` to have expired. But with this change we do.

Reply:
It used to be if you have 0 seconds left on your clock, you can still move until the next tick. And maybe this should have been the correct behavior all along. But it's a significant change that is neither called out here nor matches the [specification](https://specs.optimism.io/experimental/fault-proof/stage-one/fault-dispute-game.html#game-clock)

Reply:
This change should actually pull us _into_ spec, since the old version allowed for making a move 1s after half of the game duration had elapsed (note the inclusive bound of the starting timestamp).

Now, clocks have exactly `3.5 days` allocated, which makes:
> A move against a particular claim is no longer possible once the parent of the disputed claim's Clock has exceeded half of the GAME_DURATION. By which point, the claim's clock has expired.
actually hold true.